### PR TITLE
Add Intents to Documentation

### PR DESCRIPTION
**Pull request by jeffreyladd**
# Summary
Adds Intents Documentation.  This resolves part of #671 

# Details
1. Adds documentation for Intents

### CODE REVIEW

**Comment by jeffreyladd** [comment]
this change can be viewed at https://jeffreyladd.github.io/articles/beyond_basics/intents.html

**Comment by Emzi0767** [comment]
Could this be added to the migration guide?

**Comment by jeffreyladd** [comment]
I added something to the Migration Guide that points back to the article

File: docs/articles/migration/3x_to_4x.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ To proxy your traffic, create a new instance of `System.Net.WebProxy` and assign
 `VoiceNextClient`|`VoiceNextExtension`
 `BaseModule`|`BaseExtension`
 
+### Intents
+In certain circumstances intents will have to be enabled to recieved in both the
```
Comment (Emzi0767) [inline docs/articles/migration/3x_to_4x.md:23]: > to be enabled to recieved

Perhaps

> to be enabled to receive certain events

**Review by Emzi0767 — CHANGES_REQUESTED** [review]

**Review by Emzi0767 — APPROVED** [review]

**Pull request merged**